### PR TITLE
Return error for multipart file exceeding maxBytes

### PR DESCRIPTION
**Pull request by johnbrett**
First part of fix for: https://github.com/hapijs/hapi/issues/2825

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -288,6 +289,18 @@ internals.Parser.prototype.multipart = function (source, contentType) {
     next = Hoek.once(next);                                            // Modify next() for async events
     this.next = next;
 
+    if (this.settings.maxBytes !== undefined) {
+        let bytesUsed = 0;
+
+        source.on('data', (data) => {
```
Comment (Marsup) [inline lib/index.js]: Isn't this triggering old streams push mode ?

**Comment by hueniverse** [comment]
Wouldn't it be better to perform this at the pez level? This doesn't actually stops processing, just bails faster.

**Comment by cjihrig** [comment]
> This doesn't actually stops processing, just bails faster.

I haven't tested, but couldn't you just `unpipe()` the dispenser? I think you might want to do something similar on timeout too.

**Comment by hueniverse** [comment]
We need to flush our the stream but also to stop the processor from trying to finish the job. It would be much cleaner if we just passed the option down the stack instead of hacking it here.

**Comment by cjihrig** [comment]
@johnbrett moved this to https://github.com/hapijs/pez/pull/9. I'll publish a new version, and you should be able to pass `maxBytes` to the pez `Dispenser` constructor.

**Comment by johnbrett** [comment]
I'd meant to ask how we'd tackle this today :) Nice one @cjihrig, ready to make the update when the updated `pez` version is published

**Comment by cjihrig** [comment]
OK, pez@2.1.0 is out.

**Pull request merged**